modules/ossm-deploying-jaeger.adoc

```diff
@@ -0,0 +1,91 @@
++
+[NOTE]
+====
+In-memory storage is not persistent, which means that if the Jaeger instance shuts down, restarts, or is replaced, that your trace data will be lost.  And in-memory storage cannot be scaled, since each pod has its own memory. For persistent storage, you must use the `production` or `streaming` strategies, which use Elasticsearch as the default storage.
+====
+
+* *production* - The production strategy is intended for production environments, where long term storage of trace data is important, as well as a more scalable and highly available architecture is required. Each of the backend components is therefore deployed separately.  The Agent can be injected as a sidecar on the instrumented application or as a daemonset. The Query and Collector services are configured with a supported storage type - currently Elasticsearch. Multiple instances of each of these components can be provisioned as required for performance and resilience purposes.
+
+* *streaming* - The streaming strategy is designed to augment the production strategy by providing a streaming capability that effectively sits between the Collector and the backend storage (Elasticsearch). This provides the benefit of reducing the pressure on the backend storage, under high load situations, and enables other trace post-processing capabilities to tap into the real time span data directly from the streaming platform (https://access.redhat.com/documentation/en-us/red_hat_amq/7.6/html/using_amq_streams_on_openshift/index[AMQ Streams]/ https://kafka.apache.org/documentation/[Kafka]).
+
+[NOTE]
+====
+The streaming strategy requires an additional Red Hat subscription for AMQ Streams.
+====
+
+
+If you do not specify Jaeger configuration options, the Control Plane will use the `allInOne` Jaeger deployment strategy.  You can accept the defaults or specify additional configuration options under `install`.
+
+.Control Plane default Jaeger parameters (Memory)
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+spec:
+  version: v2.0
+  tracing:
+    sampling: 10000
+    type: Jaeger
+  addons:
+    jaeger:
+      name: jaeger
+      install:
+        storage:
+          type: Memory
+----
+
+To use the `production` deployment strategy, set  `spec:addons:jaeger:install:storage:type` to 'Elasticsearch' and specify additional configuration options under `install`.   Or you can create and configure your Jaeger instance and set  `spec:addons:jaeger:name` to the name of the Jaeger instance, for example  `jaeger-production`.
+
+.Control Plane default Jaeger parameters (Elasticsearch)
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: production
+spec:
+  version: v2.0
+  tracing:
+    sampling: 10000
+    type: Jaeger
+  addons:
+    jaeger:
+      name: jaeger
+      install:
+        storage:
+          type: Elasticsearch
```

Comment:
what about the number of ES nodes, resource rquirements, etc., as discussed in the other doc above?

Reply:
These would be configured under spec.addons.jaeger.install.storage.elasticsearch, along with resource settings under spec.runtime.components.

Reply:
might need to have an example showing these @JStickler

Reply:
Here are the supported fields for elasticsearch storage: https://github.com/rcernich/istio-operator/blob/apidocgen/docs/crd/maistra.io_ServiceMeshControlPlane_AddonsConfig_v2.adoc#JaegerElasticsearchStorageConfig

The resources are more complicated, but I think those are already documented.  The section would look something like: (note, not all runtime fields are supported by the jaeger charts)
```yaml
spec:
  runtime:
    components:
      tracing.jaeger.elasticsearch: # only supports resources and image name
        container:
          resources: {}
      tracing.jaeger.agent: {} # only supports overriding image name
      tracing.jaeger.allInOne: {} # only supports overriding image name
      tracing.jaeger.collector: {} # only supports overriding image name
      tracing.jaeger.query: {} # only supports overriding image name
      tracing.jaeger: {} # maps to jaeger spec.resources, as well as jaeger specific hub/tag, and annotations
```

Reply:
@rcernich The image names should never be overridden, as they would not be supported - so is there a benefit of having these entries?

Reply:
Nope.  I was just mentioning them for completeness, as they are in the charts, so I assume they were added for some reason.  Given that, probably makes sense to just highlight how resources can be configured.

Reply:
OK, I hope I'm interpreting this thread correctly.  I added the first part of Rob's example (from spec down to resources {}) but since we don't want customers to change the image names, I did not include the last five lines.